bleak/backends/dotnet/scanner.py

```diff
@@ -89,9 +90,23 @@ def _received_handler(self, sender, event_args):
                 if event_args.BluetoothAddress not in self._devices:
                     self._devices[event_args.BluetoothAddress] = event_args
         if self._callback is not None:
-            self._callback(sender, event_args)
+            # Get a "BLEDevice" from parse_event args
+            temporary_device = self.parse_eventargs(event_args)
+
+            # Use the BLEDevice to populate all the fields for the advertisement data to return
+            advertisement_data = AdvertisementData(
+                address=temporary_device.address,
+                local_name=temporary_device.name or "Unknown",
+                rssi=temporary_device.rssi,
+                manufacturer_data=temporary_device.metadata["manufacturer_data"],
+                service_data=event_args.Advertisement.GetSectionsByType(0x16),
```

Comment:
The service data needs to be parsed to put it in the same format as other platforms. Here is what I came up with:

```python
            # Decode service data
            service_data = {}
            # 0x16 is service data with 16-bit UUID
            for section in event_args.Advertisement.GetSectionsByType(0x16):
                with BleakDataReader(section.Data) as reader:
                    data = reader.read()
                    service_data[
                        f"0000{data[1]:02x}{data[0]:02x}-0000-1000-8000-00805f9b34fb"
                    ] = data[2:]
            # 0x20 is service data with 32-bit UUID
            for section in event_args.Advertisement.GetSectionsByType(0x20):
                with BleakDataReader(section.Data) as reader:
                    data = reader.read()
                    service_data[
                        f"{data[3]:02x}{data[2]:02x}{data[1]:02x}{data[0]:02x}-0000-1000-8000-00805f9b34fb"
                    ] = data[4:]
            # 0x21 is service data with 128-bit UUID
            for section in event_args.Advertisement.GetSectionsByType(0x21):
                with BleakDataReader(section.Data) as reader:
                    data = reader.read()
                    service_data[str(UUID(bytes_le=data[:16]))] = data[16:]
```

Reply:
Thank you for your help with this!